var/spack/repos/builtin/packages/leveldb/package.py

```diff
@@ -7,40 +7,93 @@
 from spack import *
 
 
-class Leveldb(MakefilePackage):
+class Leveldb(CMakePackage):
     """LevelDB is a fast key-value storage library written at Google
     that provides an ordered mapping from string keys to string values."""
 
     homepage = "https://github.com/google/leveldb"
-    url      = "https://github.com/google/leveldb/archive/v1.20.tar.gz"
+    url      = "https://github.com/google/leveldb/archive/1.22.tar.gz"
     git      = "https://github.com/google/leveldb.git"
 
     version('master', branch='master')
+    version('1.22', '55423cac9e3306f4a9502c738a001e4a339d1a38ffbee7572d4a07d5d63949b2')
     version('1.20', '298b5bddf12c675d6345784261302252')
     version('1.18', '73770de34a2a5ab34498d2e05b2b7fa0')
 
-    depends_on("snappy")
+    variant('shared', default=True, description='Build shared library')
 
+    depends_on('cmake@3.9:', when='@1.21:', type='build')
+
+    depends_on('snappy')
+
+    def url_for_version(self, version):
+        url = 'https://github.com/google/leveldb/archive/{0}.tar.gz'
+
+        if version >= Version('1.21'):
+            ver = version
+        else:
+            ver = 'v{0}'.format(version)
+
+        return url.format(ver)
+
+    @property
+    def libs(self):
+        shared = '+shared' in self.spec
+
+        return find_libraries('libleveldb', root=self.prefix,
+                              shared=shared, recursive=True)
```

Comment:
Isn't this equivalent to the default `libs` implementation?

Reply:
I thought so as well but if I remove it, the build fails with:
```
==> Error: AttributeError: 'Leveldb' object has no attribute 'libs'
```
It seems there is no default implementation? A few other packages also seem to implement it like this (for example, zlib).

Reply:
Hmm, this definitely used to work. Did anything change recently @alalazo? The reason that `zlib` has its own `libs` property is because the default would look for `libzlib`, not `libz`. Maybe this is only required when you are trying to access the `libs` property from the same package?

Reply:
> The reason that `zlib` has its own `libs` property is because the default would look for `libzlib`, not `libz`.

Ah, got it.

> Maybe this is only required when you are trying to access the `libs` property from the same package?

After playing around with it some more, `self.spec['leveldb'].libs` does work. :man_shrugging: I have pushed an updated version without the `libs` method.